packages/grid/data-grid/src/tests/filtering.DataGrid.test.tsx

```diff
@@ -0,0 +1,106 @@
+import * as React from 'react';
+import {
+  createClientRenderStrictMode,
+} from 'test/utils/index';
+import { expect } from 'chai';
+import { DataGrid } from '@material-ui/data-grid';
+import { getColumnValues, raf, sleep } from 'test/utils/helperFn';
+
+describe('<DataGrid />', () => {
+  // TODO v5: replace with createClientRender
+  const baselineProps = {
+    rows: [
+      {
+        id: 0,
+        brand: 'Nike',
+      },
+      {
+        id: 1,
+        brand: 'Adidas',
+      },
+      {
+        id: 2,
+        brand: 'Puma',
+      },
+    ],
+    columns: [{field: 'brand'}],
+  };
+
+  describe('filter', () => {
+    let setProps: any;
+
+    before(function beforeHook() {
+      if (/jsdom/.test(window.navigator.userAgent)) {
+        // Need layouting
+        this.skip();
+      }
+    });
+
+    const TestCase = (props: {  operator?: string, value?: string }) => {
+      const {operator, value} = props;
+      return (
+        <div style={{width: 300, height: 300}}>
+          <DataGrid
+            {...baselineProps}
+            filterModel={{
+              items: [{
+                columnField: 'brand', value, operatorValue: operator
+              }]
+            }}
+            disableColumnFilter={false}
+          />
+        </div>
+      );
+    };
+    beforeEach(()=> {
+      const render = createClientRenderStrictMode();
+      const renderResult = render(<TestCase value={'a'} operator={'contains'}/>);
+      setProps = renderResult.setProps;
```

Comment:
Please don't, best to duplicate the work between each test. Abstraction in tests can very quickly be harmful, I think that we need to see a lot of duplication before considering abstracting (I would but a higher threshold for tests than for actual code)